### PR TITLE
loadbalancer: Remove error handling of protocol string

### PR DESCRIPTION
**Pull request by tgraf**
The protocol value is currently ignored so all of this error handling is only
complicating the caller logic.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6072)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

File: pkg/loadbalancer/loadbalancer.go [hunk 1]
```diff
@@ -305,13 +305,8 @@ type L4Addr struct {
...
 
 // NewL4Addr creates a new L4Addr. Returns an error if protocol is not recognized.
-func NewL4Addr(protocol L4Type, number uint16) (*L4Addr, error) {
-	switch protocol {
-	case TCP, UDP, NONE:
-	default:
-		return nil, fmt.Errorf("unknown protocol type %s", protocol)
-	}
-	return &L4Addr{Protocol: protocol, Port: number}, nil
+func NewL4Addr(protocol L4Type, number uint16) *L4Addr {
```
Comment (ianvernon) [inline pkg/loadbalancer/loadbalancer.go:308]: Update documentation to not reference error.

**Review by ianvernon — COMMENTED** [review]
LGTM apart from needed doc change, which when fixed, does not need another CI run.

**Pull request merged**